src/main/java/com/snowflake/kafka/connector/Utils.java

```diff
@@ -440,11 +469,53 @@ && parseTopicToTableMap(config.get(SnowflakeSinkConnectorConfig.TOPICS_TABLES_MA
           Utils.formatString("{} cannot be empty.", SnowflakeSinkConnectorConfig.SNOWFLAKE_SCHEMA));
     }
 
-    if (!config.containsKey(SnowflakeSinkConnectorConfig.SNOWFLAKE_PRIVATE_KEY)) {
-      invalidConfigParams.put(
-          SnowflakeSinkConnectorConfig.SNOWFLAKE_PRIVATE_KEY,
-          Utils.formatString(
-              "{} cannot be empty.", SnowflakeSinkConnectorConfig.SNOWFLAKE_PRIVATE_KEY));
+    switch (config.getOrDefault(
+        SnowflakeSinkConnectorConfig.AUTHENTICATOR_TYPE, Utils.SNOWFLAKE_JWT)) {
+        // TODO: SNOW-889748 change to enum
+      case Utils.SNOWFLAKE_JWT:
```

Comment:
Will it work if I have uppercase in the configuration?

Reply:
Added `tolower()` before entering switch.

Reply:
Thanks, could we add a unit test for this?

Reply:
Added